aiosmtpd/smtp.py

```diff
@@ -438,7 +438,12 @@ def smtp_DATA(self, arg):
                 'mail_options': self.mail_options,
                 'rcpt_options': self.rcpt_options,
             }
-        status = self.event_handler.process_message(*args, **kwargs)
+        if asyncio.iscoroutinefunction(self.event_handler.process_message):
+            kwargs['loop'] = self.loop
+            status = yield from self.event_handler.process_message(
+                *args, **kwargs)
+        else:
+            status = self.event_handler.process_message(*args, **kwargs)
```

Comment:
Thanks for the answer.  Do you think we should just make `process_message()` a coroutine in all cases?  That's an API change but I think it would be okay in this case.